Sources/SourceKitLSP/Rename.swift

```diff
@@ -500,7 +500,7 @@ extension SourceKitLSPServer {
   ) async -> (swiftLanguageService: SwiftLanguageService, snapshot: DocumentSnapshot, location: SymbolLocation)? {
     var reference: SymbolOccurrence? = nil
     index.forEachSymbolOccurrence(byUSR: usr, roles: renameRoles) {
-      if index.symbolProvider(for: $0.location.path) == .swift {
+      if index.unchecked.symbolProvider(for: $0.location.path) == .swift {
```

Comment:
Doesn't the reference here need to be up to date?

Reply:
`index.forEachSymbolOccurrence` only returns valid references because `index` is a `CheckedIndex`. 

Once we have the reference, it doesn’t make sense to find the symbol provider for a document with respect to a `CheckedIndex`:
- Checking for `.deletedFile` doesn’t make sense because we wouldn’t even have a reference from within the file. And if you ask for the symbol provider of a modified file: Sure, why not just return the last language that we know about.
- Checking for `.modifiedFile` or `.inMemoryModifiedFiles` also doesn’t really make sense because it’s better to get the last known symbol provider than no symbol provider at all. A source file’s symbol provider doesn’t change just because the file is modified on disk.